app/controllers/api/api_controller.rb

```diff
@@ -118,44 +98,32 @@ def render_expired_token_error
         tag: 'expired-access-token',
         message: 'The provided access token has expired.',
       },
-    }, status: :expired_access_token
+    }, status: EXPIRED_TOKEN_HTTP_CODE
   end
 
-  def token_from_request_header
-    unless request.headers['Authorization'].present?
-      return
-    end
-
-    strategy, token = request.headers['Authorization'].split(' ')
-
-    if (strategy || '').downcase != 'bearer'
-      return
-    end
-
-    token
-  end
-
-  def user_from_authentication(authentication)
-    return authentication[:session].user unless authentication[:session].nil?
-
-    User.find_by_uuid authentication[:user_uuid]
-  end
-
-  def validate_token(token)
-    # Try JWT first...
+  def decode_token(token)
+    # Try JWT first
     claims = begin
               SyncEngine::JwtHelper.decode(token)
              rescue
                nil
             end
 
-    return { type: 'jwt', user_uuid: claims['user_uuid'], claims: claims } unless claims.nil?
-
-    # See if it's an access_token...
-    session = Session.where(access_token: token).first
-
-    if session && ActiveSupport::SecurityUtils.secure_compare(token, session.access_token)
```

Comment:
`secure_compare` seems redundant here. This will always be true, since we found the Session corresponding to this token.